main.tf

```diff
@@ -43,15 +43,16 @@ resource "google_container_node_pool" "new_container_cluster_node_pool" {
   #}
 
   management {
-    auto_repair  = "${lookup(var.node_pool, "auto_repair", true)}"
-    auto_upgrade = "${lookup(var.node_pool, "auto_upgrade", true)}"
+    auto_repair  = "${lookup(var.node_pool[count.index], "auto_repair", true)}"
+    auto_upgrade = "${lookup(var.node_pool[count.index], "auto_upgrade", true)}"
   }
 
-  lifecycle {
-    "ignore_changes" = ["version"]
-  }
+  #lifecycle {
+  #  "ignore_changes" = ["version"]
```

Comment:
I remember there being some issue with tracking the version in terraform but I don't remember what. I'll do some poking around again and let you know.

Reply:
Okay, @luhkevin. Here we should track the node version via terraform templates.